web/src/hotkey.js

```diff
@@ -410,11 +410,12 @@ function handle_popover_events(event_name) {
 
 // Returns true if we handled it, false if the browser should.
 export function process_enter_key(e) {
-    if ($(".dropdown.open, .dropup.open").length > 0 && $(e.target).attr("role") === "menuitem") {
-        // on dropdown menu elements, force a click and prevent default.
-        // this is because these links do not have an href and so don't force a
-        // default action.
+    if (popovers.any_active() && $(e.target).attr("role") === "menuitem") {
+        // If a popover is open and we pressed Enter on a menu item,
+        // call click directly on the item to navigate to the `href`.
+        // trigger("click") doesn't work for them to navigate to `href`.
         e.target.click();
+        e.preventDefault();
         return true;
```

Comment:
Opened discussion for this https://chat.zulip.org/#narrow/stream/6-frontend/topic/Navigating.20to.20href.20of.20a.20tags.20in.20popovers/near/1662293